sagan-common/src/main/java/sagan/projects/ProjectRelease.java

```diff
@@ -183,44 +187,75 @@ public ProjectRelease createWithVersionPattern() {
         return release;
     }
 
+    private static final Map<String, Integer> SPECIAL_MEANINGS =
+            ImmutableMap.of("dev", -1, "rc", 1, "release", 2, "final", 3);
+
+    /**
+     * Adapted from Gradle's StaticVersionComparator
+     */
     @Override
     public int compareTo(ProjectRelease other) {
-        if (this.getVersion() == null || other.getVersion() == null) {
-            return 0;
-        }
-
-        Pattern pattern = Pattern.compile("([0-9]+)");
+        if(other == null) return -1;
 
-        Matcher thisMatcher = pattern.matcher(this.getVersion());
-        Matcher otherMatcher = pattern.matcher(other.getVersion());
+        Version version1 = Version.build(getVersion());
+        Version version2 = Version.build(other.getVersion());
 
```

Comment:
Instead of "version1"/"version2", can we use something more semantic, like "thisVersion"/"otherVersion"?